### PR TITLE
Switch to using a function for data inserts

### PR DESCRIPTION
**Pull request by cevian**
Previously, we used INSERT INTO for copying data into data tables.
This, however requires us to prepare a separate statement for each
unique metric. This ended up creating a lot of prepared statements
(# metrics * # copiers). Each prepare has an overhead of at least
one RTT. This is a problem in environments with high latency.

We switch to using a function that takes the metric name as a parameter
This reduces number of prepared statements for the copier to 1.

We also reduce the statement cache capacity back to the PGX default
of 512.

### CODE REVIEW

File: pkg/pgclient/client.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func getPgConfig(cfg *Config) (*pgxpool.Config, int, error) {
 	} else {
 		connectionArgsFmt = "%s&pool_max_conns=%d&pool_min_conns=%d&statement_cache_capacity=%d"
 	}
-	statementCacheCapacity := cfg.CacheConfig.MetricsCacheSize * 2
+	statementCacheCapacity := 512
```
Comment (antekresic) [inline pkg/pgclient/client.go:92]: We should add a comment at this number so we know why we chose it.

File: pkg/pgmodel/ingestor/insert.go [hunk 2]
```diff
@@ -385,13 +383,14 @@ func doInsert(conn pgxconn.PgxConn, reqs ...copyRequest) (err error) {
...
 	var affectedMetrics uint64
 	for _, numRows := range numRowsPerInsert {
-		ct, err := results.Exec()
+		var insertedRows int
+		err := results.QueryRow().Scan(&insertedRows)
 		if err != nil {
 			return err
 		}
-		if int64(numRows) != ct.RowsAffected() {
+		if int64(numRows) != int64(insertedRows) {
```
Comment (antekresic) [inline pkg/pgmodel/ingestor/insert.go:391]: Minor nit: maybe we can simplify these var types so we have to juggle less. I haven't looked at the code too deeply to check if its possible/easy.

**Review by antekresic — APPROVED** [review]
Looks good, couple of minor things but its ready for merge even like this.

**Comment by VineethReddy02** [comment]
LGTM!

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**